registry/remote/repository.go

```diff
@@ -806,6 +806,10 @@ func (s *blobStore) Mount(ctx context.Context, desc ocispec.Descriptor, fromRepo
 	var r io.ReadCloser
 	if getContent != nil {
 		r, err = getContent()
+		if errors.Is(err, errdef.ErrUnsupported) {
+			// getContent can return a ErrUnsupported to fallback to the default copy operation
+			r, err = s.sibling(fromRepo).Fetch(ctx, desc)
+		}
```

Comment:
This behavior of `errdef.ErrUnsupported` should also be documented in https://pkg.go.dev/oras.land/oras-go/v2@v2.3.1/registry#Mounter

Reply:
To avoid breaking change and confusion, I would like to introduce an internal error instance to replace the well known  `errdef.ErrUnsupported` so that users won't get surprised when they do
```go
store.Mount(ctx, desc, fromRepo, func() (io.ReadCloser, error) {
    return nil, errdef.ErrUnsupported
})

Reply:
Do you really think the error should be non-exported (internal)?  I think it might be better to be exported so consumers of the library can use it.  `clone` and `sibling` are not exported so user's defining your own Fetch based on the existing repo is slightly error prone.  But we can decide this later, if we can implement the more complicated use case (trying multiple mounts within a copy for a single blob) then we can easily justify keeping this sentinel error internal.

So what should the name be?  Maybe `unimplemented` or `useSourceRepository`.  Also where would we want this defined.  For now I exported it and defined it right next to the definition of the `type Mounter interface`.  If we make it internal (but exported from the package because it is needed in both the top level package and `registry/remote` we probably need to define a new internal package at the top level.